ckanext/qa/tasks.py

```diff
@@ -395,6 +395,16 @@ def save_qa_result(resource_id, qa_result, log):
     qa.archival_timestamp == qa_result['archival_timestamp']
     qa.updated = now
 
+    # If the resource doesn't have a format, use the one we discovered in QA.
+    if not resource.format and qa.format:
+        rev = model.repo.new_revision()
+        rev.author = u'QA Task'
```

Comment:
Could you make this 'script-qa', please? It's just that when we generate this report: https://data.gov.uk/data/report/publisher-activity/cabinet-office we ignore any changes done by an author starting 'script', such as these ones.